ansible/roles/ocp4_workload_integreatly/tasks/workload.yml

```diff
@@ -0,0 +1,208 @@
+  register: _create_bucket
+  retries: 20
+  delay: 60
+  failed_when: _create_bucket.rc != 0 and 'BucketAlreadyOwnedByYou' not in _create_bucket.stderr
+  until: _create_bucket.rc == 0 or 'BucketAlreadyOwnedByYou' in _create_bucket.stderr
+
+- name: Create Minio blobstorage override
+  k8s:
+    state: present
+    definition:
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: threescale-blobstorage-{{ ocp4_workload_integreatly_custom_resource_name }}
+        namespace: "{{ ocp4_workload_integreatly_namespace }}"
+      data:
+        bucketName: "{{ ocp4_workload_integreatly_minio_bucket_name | b64encode }}"
+        bucketRegion: "{{ ocp4_workload_integreatly_minio_bucket_region | b64encode }}"
+        credentialKeyID: "{{ ocp4_workload_integreatly_minio_access_key | b64encode }}"
+        credentialSecretKey: "{{ ocp4_workload_integreatly_minio_secret_key | b64encode }}"
+        AWS_HOSTNAME: "{{ _minio_route.resources[0].spec.host | b64encode }}"
+        AWS_PATH_STYLE: "{{ true | b64encode }}"
+
+- set_fact:
+    _format_vars:
+      release: "{{ ocp4_workload_integreatly_release }}"
+      branch: "{{ ocp4_workload_integreatly_git_ref }}"
+
+- set_fact:
+    _operator_manifest_dir: "{{ ocp4_workload_integreatly_manifest_dir_format.format(**_format_vars) }}"
+
+# Retrieve
+- set_fact:
+    _operator_manifest_csv_paths: "{{ (_operator_manifest_csv_paths | default([])) + [lookup(ocp4_workload_integreatly_manifest_dir_type, _operator_manifest_dir + '/' + item.format(**_format_vars), split_lines=False) | from_yaml] }}"
+  with_items: "{{ ocp4_workload_integreatly_manifest_csv_formats }}"
+
+# Retrieve Packages
+- set_fact:
+    _operator_manifest_package_paths: "{{ (_operator_manifest_package_paths | default([])) + [lookup(ocp4_workload_integreatly_manifest_dir_type, _operator_manifest_dir + '/' + item.format(**_format_vars), split_lines=False) | from_yaml] }}"
+  with_items: "{{ ocp4_workload_integreatly_manifest_package_formats }}"
+
+
+# Retrieve CRDS
+- set_fact:
+    _operator_manifest_crd_paths: "{{ (_operator_manifest_crd_paths | default([])) + [lookup(ocp4_workload_integreatly_manifest_dir_type, _operator_manifest_dir + '/' + item.format(**_format_vars), split_lines=False) | from_yaml] }}"
+  with_items: "{{ ocp4_workload_integreatly_manifest_crd_formats }}"
+
+# By default the Integreatly operator will create it's own RHMI resource if one
+# isn't found in the namespace. Create the CRDs before OLM to allow an RHMI
+# resource to be created before the operator starts up.
+- name: Create expected CRDs to ensure automated RHMI CR isn't created on operator startup
+  k8s:
+    state: present
+    namespace: "{{ ocp4_workload_integreatly_namespace }}"
+    merge_type:
+    - strategic-merge
+    - merge
+    definition: "{{ item }}"
+  loop: "{{ _operator_manifest_crd_paths }}"
+
+- name: Create workshop RHMI resource
```

Comment:
@aidenkeating Is there a need to validate that Minio is available before trying to install an instance of RHMI? Or is it a case that by the time the integreatly operator gets around to configuring 3Scale, Minio would have had plenty of time to come up by then?

Reply:
The Minio workload should not exit cleanly until minio is deployed OK.

Reply:
@pmccarthy Minio is deployed successfully by the time we reach this stage